### ISSUE TITLE
Did you mean masked_fill_ in modeling_deberta.py?

### ISSUE THREAD

**Issue by xadupre** [huggingface/transformers#35162]
### System Info

- `transformers` version: 4.47.0
- Platform: Linux-5.15.167.4-microsoft-standard-WSL2-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface_hub version: 0.24.6
- Safetensors version: 0.4.4
- Accelerate version: 1.0.0
- Accelerate config:    not found
- PyTorch version (GPU?): 2.6.0.dev20241205+cu124 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: <fill in>
- Using GPU in script?: <fill in>
- GPU type: NVIDIA GeForce RTX 4060 Laptop GPU

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

masked_fill is called but the result is not used: https://github.com/huggingface/transformers/blob/main/src/transformers/models/deberta/modeling_deberta.py#L293.

The same issue happens in modeling_deberta_v2.py.

### Expected behavior

Use masked_fill_ instead.

**Comment by xadupre** [comment]
I'll do that. Any recommandation for a unit test to add, any existing good one I could leverage?

**Comment by Rocketknight1** [comment]
@xadupre The ideal test here would load or initialize a small model, then create an attention mask, and then verify using `output_attentions` that attentions correctly follow the mask. Take a look at the tests in `tests/models/deberta/` and see if there's one you can copy/modify for your needs, and feel free to ping me for reviews or advice once the PR is made.

**Comment by xadupre** [comment]
I finally removed the lines. I give some explanations in the PR but using inplace masked_fill would set all values in attention_scores to zero. Maybe the code was refactored and these lines were forgotten.
